verifier/credentialsConfig.go

```diff
@@ -53,39 +51,71 @@ func InitServiceBackedCredentialsConfig(repoConfig *config.ConfigRepo) (credenti
 			logging.Log().Warnf("Was not able to instantiate the config client.")
 		}
 	}
-	var serviceCache Cache = cache.New(CACHE_EXPIRY*time.Second, 2*CACHE_EXPIRY*time.Second)
 
-	scb := ServiceBackedCredentialsConfig{configClient: &configClient, serviceCache: serviceCache, initialConfig: repoConfig}
+	scb := ServiceBackedCredentialsConfig{configClient: &configClient, initialConfig: repoConfig}
+
+	err = scb.fillStaticValues()
+	if err != nil {
+		return nil, err
+	}
 
-	scb.fillStaticValues()
 	if repoConfig.ConfigEndpoint != "" {
-		chrono.NewDefaultTaskScheduler().ScheduleAtFixedRate(scb.fillCache, time.Duration(30)*time.Second)
+		_, err := chrono.NewDefaultTaskScheduler().ScheduleAtFixedRate(scb.fillCache, time.Duration(30)*time.Second)
+		if err != nil {
+			logging.Log().Errorf("failed scheduling task: %v", err)
+			return nil, err
+		}
 	}
 
 	return scb, err
 }
 
-func (cc ServiceBackedCredentialsConfig) fillStaticValues() {
+func (cc ServiceBackedCredentialsConfig) fillStaticValues() error {
 	for _, configuredService := range cc.initialConfig.Services {
 		logging.Log().Debugf("Add to service cache: %s", configuredService.Id)
-		cc.serviceCache.Add(configuredService.Id, configuredService, cache.NoExpiration)
+		err := common.GlobalCache.ServiceCache.Add(configuredService.Id, configuredService, cache.DefaultExpiration)
+		if err != nil {
+			logging.Log().Errorf("failed caching configured service %s in fillStaticValues(): %v", configuredService.Id, err)
+			return err
+		}
 	}
+	return nil
 }
 
-func (cc ServiceBackedCredentialsConfig) fillCache(ctx context.Context) {
-	client := *(cc.configClient)
-	services, err := client.GetServices()
+func (cc ServiceBackedCredentialsConfig) fillCache(context.Context) {
+	configClient := *(cc.configClient)
+	services, err := configClient.GetServices()
 	if err != nil {
 		logging.Log().Warnf("Was not able to update the credentials config from the external service. Will try again. Err: %v.", err)
 		return
 	}
 	for _, configuredService := range services {
-		cc.serviceCache.Add(configuredService.Id, configuredService, cache.NoExpiration)
+		err := common.GlobalCache.ServiceCache.Add(configuredService.Id, configuredService, cache.DefaultExpiration)
+		if err != nil {
+			logging.Log().Errorf("failed caching configured service in fillCache(): %v", err)
+		}
+
+		var tirEndpoints []string
+
+		for serviceScope, credentials := range configuredService.ServiceScopes {
+			for _, credential := range credentials {
+				serviceIssuersLists, err := cc.GetTrustedIssuersLists(configuredService.Id, serviceScope, credential.Type)
+				if err != nil {
+					logging.Log().Errorf("failed caching issuers lists in fillCache(): %v", err)
+				}
+				tirEndpoints = append(tirEndpoints, serviceIssuersLists...)
```

Comment:
Why do we append if there was an error? I theory serviceIssuersLists should be empty and not cause any harm, but still